tests/provision/provisioning_rules.go

```diff
@@ -0,0 +1,1708 @@
+	node3lbl := labelDict{"media_type": mediaSsd, "vps_test": "test"}
+	node4lbl := labelDict{"media_type": mediaSsd, "vps_test": "test"}
+	lbldata = append(lbldata, node1lbl, node2lbl, node3lbl, node4lbl)
+	return lbldata, 1
+}
+
+//StorageClass placement_strategy mapping
+func (v *vpscase2) GetScStrategyMap() map[string]string {
+	return map[string]string{"placement-1": "placement-1", "placement-2": "placement-2", "placement-3": ""}
+}
+
+func (v *vpscase2) GetSpec() string {
+
+	var vpsSpec string
+	vpsSpec = `apiVersion: portworx.io/v1beta2
+kind: VolumePlacementStrategy
+metadata:
+  name: placement-1
+spec:
+  replicaAffinity:
+  - enforcement: preferred
+    matchExpressions:
+    - key: media_type
+      operator: In
+      values:
+      - "SSD"
+---
+apiVersion: portworx.io/v1beta2
+kind: VolumePlacementStrategy
+metadata:
+  name: placement-2
+spec:
+  replicaAffinity:
+  - enforcement: preferred
+    matchExpressions:
+    - key: media_type
+      operator: In
+      values:
+      - "SSD"`
+	return vpsSpec
+}
+
+func (v *vpscase2) CleanVps() {
+	logrus.Infof("Cleanup test case context for: %v", v.name)
+}
+
+//#---- Case 3 ----T809561: Verify Lt, Gt operators using latency and iops
+type vpscase3 struct {
+	//Case description
+	name string
+	// Enabled
+	enabled bool
+}
+
+func (v *vpscase3) GetLabels() ([]labelDict, int) {
+
+	lbldata := []labelDict{}
+	node1lbl := labelDict{"iops": "90", "latency": "50"}
+	node2lbl := labelDict{"iops": "80", "latency": "40"}
+	node3lbl := labelDict{"iops": "70", "latency": "30"}
+	node4lbl := labelDict{"iops": "60", "latency": "20"}
+	lbldata = append(lbldata, node1lbl, node2lbl, node3lbl, node4lbl)
+	return lbldata, 1
+}
+
+//StorageClass placement_strategy mapping
+func (v *vpscase3) GetScStrategyMap() map[string]string {
+	return map[string]string{"placement-1": "placement-1", "placement-2": "placement-2", "placement-3": ""}
+}
+
+func (v *vpscase3) GetSpec() string {
+
+	var vpsSpec string
+	vpsSpec = `apiVersion: portworx.io/v1beta2
+kind: VolumePlacementStrategy
+metadata:
+  name: placement-1
+spec:
+  replicaAffinity:
+  - enforcement: required
+    matchExpressions:
+    - key: iops
+      operator: Gt
+      values:
+      - "60"
+---
+apiVersion: portworx.io/v1beta2
+kind: VolumePlacementStrategy
+metadata:
+  name: placement-2
+spec:
+  replicaAffinity:
+  - enforcement: required
+    matchExpressions:
+    - key: latency
+      operator: Lt
+      values:
+      - "50"`
+	return vpsSpec
+}
+
+func (v *vpscase3) CleanVps() {
+	logrus.Infof("Cleanup test case context for: %v", v.name)
+}
+
+//#---- Case 4 ----T863792  Verify Replica Affinity with topology keys
+type vpscase4 struct {
+	//Case description
+	name string
+	// Enabled
+	enabled bool
+}
+
+func (v *vpscase4) GetLabels() ([]labelDict, int) {
+
+	lbldata := []labelDict{}
+	node1lbl := labelDict{"failure-domain.beta.kubernetes.io/px_zone": "east", "failure-domain.beta.kubernetes.io/px_region": "usa"}
```

Comment:
I see this code is being repeated a lot, is there any way we can improve it?

Reply:
For each test case, the  kubernetes nodes are labelled differently  and hence you are seeing this code with different values or  different combination of entry

Reply:
It's okay for each test case to have it's own labels